README.md
rrdtool Cookbook
================

[![Build Status](https://travis-ci.org/vkhatri/chef-rrdtool.svg?branch=master)](https://travis-ci.org/vkhatri/chef-rrdtool)

This is a [Chef] cookbook to manage [RRDTool] and rrdcached.


## Repository

https://github.com/vkhatri/chef-rrdtool


## Recipes

- `rrdtool::default`      - default recipe (used for run_list)

- `rrdtool::config`      - configures rrdtool for rrdcached

- `rrdtool::install`      - install rrdtool


## Attributes

* `default['rrdtool']['packages']` (default: `calculated`): rrdtool packages

* `default['rrdtool']['setup_rrdcached']` (default: `true`): whether to setup rrdcached

* `default['rrdtool']['user']` (default: `nagios`): rrdtool files/socket user

* `default['rrdtool']['group']` (default: `nagios`): rrdtool files/socket group

* `default['rrdtool']['perms']` (default: `0774`): rrdtool files/socket permissions

* `default['rrdtool'][timeout'']` (default: `120`): rrdcached process parameter

* `default['rrdtool']['flush_timer']` (default: `300`): rrdcached process parameter

* `default['rrdtool']['delay']` (default: `120`): rrdcached process parameter

* `default['rrdtool']['write_threads']` (default: `4`): rrdcached process parameter

* `default['rrdtool']['rrdcached_bin']` (default: `/usr/bin/rrdcached`): rrdcached bin

* `default['rrdtool']['home_dir']` (default: `/var/rrdtool`): rrdtool base directory

* `default['rrdtool']['rrdcached_dir']` (default: `/var/rrdtool/cache`): rrdcached cache directory

* `default['rrdtool']['journal_dir']` (default: `/var/rrdtool/journal`): rrdcached jornal directory

* `default['rrdtool']['listen']` (default: `/var/rrdtool/rrdcached.sock`): rrdcached process listener

* `default['rrdtool']['options']` (default: `[]`): rrdcached process options

## Contributing

1. Fork the repository on Github
2. Create a named feature branch (like `add_component_x`)
3. Write your change
4. Write tests for your change (if applicable)
5. Run the tests (`rake`), ensuring they all pass
6. Write new resource/attribute description to `README.md`
7. Write description about changes to PR
8. Submit a Pull Request using Github


## Copyright & License

Authors:: Virender Khatri and [Contributors]

<pre>
Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
</pre>


[Chef]: https://www.chef.io/
[RRDTool]: http://oss.oetiker.ch/rrdtool/
[Contributors]: https://github.com/vkhatri/chef-rrdtool/graphs/contributors


CHANGELOG.md
rrdtool CHANGELOG
=================

This file is used to list changes made in each version of the rrdtool cookbook.

0.2.0
-----

- Virender Khatri - added rrdcached init scripts for debian platform family

- Virender Khatri - added node attribute for packages install

- Virender Khatri - added rrdcached package

0.1.7
-----

- Virender Khatri - rrdcached setup is optional now

- Virender Khatri - disabled knife temporarily in favour of travis


0.1.4
-----

- Virender Khatri - added deb family packages

- Virender Khatri - corrected typo for rrdcached -j option

- Virender Khatri - added service notify for files / templates

- Virender Khatri - added travis files


0.1.0
-----

- Virender Khatri - Initial release of rrdtool


- - -
Check the [Markdown Syntax Guide](http://daringfireball.net/projects/markdown/syntax) for help with Markdown.

The [Github Flavored Markdown page](http://github.github.com/github-flavored-markdown/) describes the differences between markdown on github and standard markdown.


recipes/config.rb
#
# Cookbook Name:: rrdtool
# Recipe:: config
#
# Copyright 2014, Virender Khatri
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

platform_sysconfig = value_for_platform_family(
  'rhel' => '/etc/sysconfig',
  'debian' => '/etc/default'
)

template ::File.join(platform_sysconfig, 'rrdcached') do
  source 'sysconfig.rrdcached.erb'
  owner 'root'
  group 'root'
  mode 0744
  notifies :restart, 'service[rrdcached]', :delayed
  variables(:group => node['rrdtool']['group'],
            :user => node['rrdtool']['user'],
            :timeout => node['rrdtool']['timeout'],
            :delay => node['rrdtool']['delay'],
            :flush_timer => node['rrdtool']['flush_timer'],
            :write_threads => node['rrdtool']['write_threads'],
            :home_dir => node['rrdtool']['home_dir'],
            :rrdcached_dir => node['rrdtool']['rrdcached_dir'],
            :listen => node['rrdtool']['listen'],
            :perms => '0660',
            :journal_dir => node['rrdtool']['journal_dir'],
            :options => node['rrdtool']['options']
           )
end

template '/etc/init.d/rrdcached' do
  source "init.rrdcached.#{node['platform_family']}.erb"
  owner 'root'
  group 'root'
  mode 0744
  notifies :restart, 'service[rrdcached]', :delayed
  variables(:home_dir => node['rrdtool']['home_dir'], :rrdcached_bin => node['rrdtool']['rrdcached_bin'])
end

service 'rrdcached' do
  supports :restart => true
  action [:enable, :start]
end


recipes/install.rb
#
# Cookbook Name:: rrdtool
# Recipe:: install
#
# Copyright 2014, Virender Khatri
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

[node['rrdtool']['home_dir'],
 node['rrdtool']['rrdcached_dir'],
 node['rrdtool']['journal_dir']
].each do |d|
  directory d do
    owner node['rrdtool']['user']
    group node['rrdtool']['group']
    mode node['rrdtool']['perms']
  end
end

node['rrdtool']['packages'].each do |p|
  package p
end


recipes/rrdcached.rb
#
# Cookbook Name:: rrdtool
# Recipe:: config
#
# Copyright 2014, Virender Khatri
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

platform_sysconfig = value_for_platform_family(
  'rhel' => '/etc/sysconfig',
  'debian' => '/etc/default'
)

template ::File.join(platform_sysconfig, 'rrdcached') do
  source 'sysconfig.rrdcached.erb'
  owner 'root'
  group 'root'
  mode 0744
  notifies :restart, 'service[rrdcached]', :delayed
  variables(:group => node['rrdtool']['group'],
            :user => node['rrdtool']['user'],
            :timeout => node['rrdtool']['timeout'],
            :delay => node['rrdtool']['delay'],
            :flush_timer => node['rrdtool']['flush_timer'],
            :write_threads => node['rrdtool']['write_threads'],
            :home_dir => node['rrdtool']['home_dir'],
            :rrdcached_dir => node['rrdtool']['rrdcached_dir'],
            :listen => node['rrdtool']['listen'],
            :perms => '0660',
            :journal_dir => node['rrdtool']['journal_dir'],
            :options => node['rrdtool']['options']
           )
  only_if { node['rrdtool']['setup_rrdcached'] }
end

template '/etc/init.d/rrdcached' do
  source "init.rrdcached.#{node['platform_family']}.erb"
  owner 'root'
  group 'root'
  mode 0744
  notifies :restart, 'service[rrdcached]', :delayed
  variables(:home_dir => node['rrdtool']['home_dir'], :rrdcached_bin => node['rrdtool']['rrdcached_bin'])
  only_if { node['rrdtool']['setup_rrdcached'] }
end

service 'rrdcached' do
  supports :restart => true
  action [:enable, :start]
  only_if { node['rrdtool']['setup_rrdcached'] }
end


attributes/default.rb

default['rrdtool']['user'] = 'nagios'
default['rrdtool']['group'] = 'nagios'
default['rrdtool']['perms'] = '0774'
default['rrdtool']['timeout'] = 120
default['rrdtool']['flush_timer'] = 300
default['rrdtool']['delay'] = 120
default['rrdtool']['write_threads'] = 4
default['rrdtool']['setup_rrdcached'] = true

default['rrdtool']['rrdcached_bin'] = '/usr/bin/rrdcached'

default['rrdtool']['home_dir'] = '/var/rrdtool'
default['rrdtool']['rrdcached_dir'] = ::File.join(node['rrdtool']['home_dir'], 'cache')
default['rrdtool']['journal_dir'] = ::File.join(node['rrdtool']['home_dir'], 'journal')

default['rrdtool']['listen'] = "unix:#{node['rrdtool']['home_dir']}/rrdcached.sock"

default['rrdtool']['options'] = []

default['rrdtool']['packages'] = value_for_platform_family(
  'debian' => %w(make rrdtool librrds-perl g++ php5-cli php5-gd libapache2-mod-php5 rrdcached),
  'rhel' => %w(php php-gd php-pdo php-common php-devel rrdtool rrdtool-devel perl-Time-HiRes libtool-ltdl rrdtool-php rrdtool-perl)
)
